File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
minus = input().split('-')

plus = []

for a in minus:
    
    sum = 0
    tmp = a.split('+')
    
    for b in tmp:
        sum += int(b) 
        
    plus.append(sum)
    
    
ans = plus[0]
for i in range(1,len(plus)):
    ans-=plus[i]
    
print(ans)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String input = br.readLine();
        String[] sub = input.split("-");
 
        int answer = mSum(sub[0]);

        for (int i = 1; i < sub.length; i++) {
            int temp = mSum(sub[i]);
            answer -= temp; 
        }
        
        System.out.println(answer);
    }
    
    static int mSum(String s) {
        int sum = 0;

        String[] temp = s.split("\\+");

        for (String num : temp) {
            sum += Integer.parseInt(num);
        }
        
        return sum;
    }
}